lib/Dialect/TTNN/IR/TTNNWorkaroundInterface.cpp

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "ttmlir/Dialect/TTNN/IR/TTNNWorkaroundInterface.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNNOps.h"
+#include "ttmlir/Dialect/TTNN/IR/TTNNWorkarounds.h"
+#include <mlir/Interfaces/DestinationStyleOpInterface.h>
+
+namespace mlir::tt::ttnn {
+#include "ttmlir/Dialect/TTNN/IR/TTNNWorkaroundInterface.cpp.inc"
+
+// Verifier function for TTNN Workaround Interface
+mlir::LogicalResult verifyTTNNWorkaroundInterface(mlir::Operation *op) {
+  TTNNWorkaroundInterface workaroundOp =
+      mlir::cast<TTNNWorkaroundInterface>(op);
+
+  // Verify that the number of input and output operand workarounds is the same
+  // as the number of tensor operands and tensor results
+  size_t tensorInputs = 0;
+  size_t tensorResults = 0;
+
+  // Count the number of tensor input operands including DPS inits
+  for (auto operand : op->getOperands()) {
+    if (mlir::isa<::mlir::RankedTensorType>(operand.getType())) {
+      tensorInputs++;
+    }
+  }
+
+  // Count the number of tensor results
+  for (auto result : op->getResults()) {
+    if (mlir::isa<::mlir::RankedTensorType>(result.getType())) {
+      tensorResults++;
+    }
+  }
+
+  TTNNOperandsWorkarounds workarounds = workaroundOp.getOperandsWorkarounds();
+
+  if (workarounds.getInputOperandWorkarounds().size() != tensorInputs) {
+    return op->emitOpError("Number of input operand workarounds does not match "
+                           "the number of tensor inputs");
+  }
+
+  if (workarounds.getOutputOperandWorkarounds().size() != tensorResults) {
+    return op->emitOpError("Number of output operand workarounds does not "
+                           "match the number of tensor results");
+  }
+
+  // For DPS ops, verify that the output workaround is the same as the input
+  // init workaround
+  if (mlir::isa<DestinationStyleOpInterface>(op)) {
+    DestinationStyleOpInterface dpsOp =
+        mlir::dyn_cast<DestinationStyleOpInterface>(op);
+
+    // Go through all the operands and for each DPS init operand, check if the
+    // output workaround is the same
+    int resultIndex = 0;
+    for (size_t i = 0; i < op->getNumOperands(); i++) {
```

Comment:
You can just walk like this and remove isDpsInit() check below
```
i = numOperands() - getNumDpsInits(); i < numOperands(); ++i
```

Reply:
Great suggestion! Thanks, will change!

Reply:
At the end, I opted for using:
dpsOp.getDpsInitOperand